### ISSUE TITLE
JavaScript

### ISSUE THREAD

**Issue by Nyight** [jacobslusser/ScintillaNET#195]
I apologize if this has already been asked, though I did look for this everywhere I could think to.

Anyway, I noticed how this project, as amazing as it is, doesn't seem to have support for javascript but that Scintilla itself seems to. Given you said that we should probably use the Scintilla docs and most things related to it for guidance, I did which pointed me towards using cpp styling for the javascript. However, after having attempted to style every cpp option with nothing being changed from within an embedded script (within the <script> tags),

I've become confused and am at a loss as to how to accomplish this, excluding writing my own html lexer to include javascript, which would require I remake, at the least, a lexer for html, css, php, asp, etc... entirely for javascript syntax highlighting. Personally this seems more like I'm simply overlooking something rather than having to go to that length... I hope.

That said, if javascript has no current support and requires rewriting the html lexer then so be it I suppose.

Thanks in advance.

**Comment by jacobslusser** [comment]
Sorry, this hasn't been fully implemented yet... but you should be able to get it working following these instructions...

For an embedded script you need to use `Lexer.Html` and the style indexes for JavaScript are in the 40 - 52 range. Here is the excerpt from the Scintilla source code:

```c++
#define SCE_HJ_START 40
#define SCE_HJ_DEFAULT 41
#define SCE_HJ_COMMENT 42
#define SCE_HJ_COMMENTLINE 43
#define SCE_HJ_COMMENTDOC 44
#define SCE_HJ_NUMBER 45
#define SCE_HJ_WORD 46
#define SCE_HJ_KEYWORD 47
#define SCE_HJ_DOUBLESTRING 48
#define SCE_HJ_SINGLESTRING 49
#define SCE_HJ_SYMBOLS 50
#define SCE_HJ_STRINGEOL 51
#define SCE_HJ_REGEX 52
```

So after you've set the lexer appropriately, instead of using ScintillaNET's `Style` constants, just use the values above. i.e.

```cs
scintilla.Styles[42].ForeColor = Color.Green; // comment
scintilla.Styles[48].ForeColor = scintilla.Styles[49].ForeColor = Color.Red; // strings
// etc...
```

For this to work I believe you need to have your JavaScript in a `<script>` tag (i.e. embedded).

For non-embedded JavaScript, use the C++ lexer.

**Comment by Nyight** [comment]
Thanks for this. I had found those constants through my searching but did not understand how to implement them into this project myself until I saw the Styles example. This is very appreciated and since you haven't had the time to implement this yourself, I've taken the liberty and adding it into the code myself, which I can paste here for you or even make a pull request. Normally I'd just make the pull request but I wasn't sure how far behind master is compared to what's on your local machine (perhaps you've already implemented it yourself already?).

Let me know if you'd like the modifications for the js support.
